### PR TITLE
Remove useless ruby_block[set-env-java-home]

### PR DESCRIPTION
**Pull request by bai**
This PR removes 'set-env-java-home' ruby_block:

* `ENV['JAVA_HOME']` within a ruby_block is scoped to that block and causes this resource to be updated on every chef run.
* More importantly: this ruby_block is not used anywhere.

cc: @erichelgeson

### CODE REVIEW

**Comment by ARentz07** [comment]
We are seeing issues with this cookbook now that this change has been merged, wherein JAVA_HOME is unset. These issues started happening about the same time as the 1.47.0 release (i.e. about 3:40 central time today). In one of our environments we are locked down to 1.46.0, and it's not experiencing the issue. 
I am going to confirm the downgrade to 1.46.0 fixes the issue, just want to give you an early heads-up since all signs point to this change being the problem.

**Comment by ARentz07** [comment]
Confirmed, changed nothing other than downgrading this cookbook and things are back to normal.

**Comment by bai** [comment]
@ARentz07 Apologies for breaking things for you. May I ask, what's your use case of ENV['JAVA_HOME'] during chef run and why you prefer it to `node['java']['java_home']`?

Asking because having a resource that's triggered on every chef run (and potentially invalidating other resources) just to assign a "global variable" seems unusual.

**Comment by ARentz07** [comment]
Part of the reason for this was that the node attribute is not guaranteed to exist based on the run_list being currently used, and we would have rather checked the setup of the system at the time. Checking the environment variable seemed like a safe "other" to trying to invoke an actual Java command for a simple sanity check (though it does not guarantee Java install).

However, we are seeing what may be due to some of the effects of Chef forking the process or the context in which it runs, since this environment variable is still invisible to us, when Java is clearly installed where we expect. We are considering just removing our check on the environment variable altogether because `command not found: java` or the like would be just fine to lead us down a path to troubleshoot.

tl;dr our earlier assumptions of checking system state vs. a node attribute were probably wrong.

As such this is not an urgent/earth-shattering issue for us.

**Comment by isuftin** [comment]
@bai On my end, I use the Java cookbook in a loosely coupled way via Berksfile for testing. I leave the implementation of how Java is installed to other cookbook authors. So I cannot guarantee that this Java cookbook will be used and prefer not to rely on using `node['java']['java_home']` in my cookbooks.
